src/cmd/services/m3coordinator/downsample/options.go

```diff
@@ -191,6 +207,287 @@ type Configuration struct {
+	// - "Mean"
+	// - "Median"
+	// - "Count"
+	// - "Sum"
+	// - "SumSq"
+	// - "Stdev"
+	// - "P10"
+	// - "P20"
+	// - "P30"
+	// - "P40"
+	// - "P50"
+	// - "P60"
+	// - "P70"
+	// - "P80"
+	// - "P90"
+	// - "P95"
+	// - "P99"
+	// - "P999"
+	// - "P9999"
+	Aggregations []aggregation.Type `yaml:"aggregations"`
+
+	// StoragePolicies is the retention/resolution storage policies to keep the
+	// matched metrics at.
+	StoragePolicies []StoragePolicyConfiguration `yaml:"storagePolicies"`
+
+	// Drop specifies to drop any metrics that match the filter rather than
+	// keeping them with a storage policy.
+	Drop bool `yaml:"drop"`
+
+	// Optional fields follows.
+
+	// Name is optional.
+	Name string `yaml:"name"`
+}
+
+// Rule returns the mapping rule for the mapping rule configuration.
+func (r MappingRuleConfiguration) Rule() (view.MappingRule, error) {
+	id := uuid.New()
+	name := r.Name
+	if name == "" {
+		name = id
+	}
+	filter := r.Filter
+
+	aggID, err := aggregation.CompressTypes(r.Aggregations...)
+	if err != nil {
+		return view.MappingRule{}, err
+	}
+
+	storagePolicies, err := StoragePolicyConfigurations(r.StoragePolicies).StoragePolicies()
+	if err != nil {
+		return view.MappingRule{}, err
+	}
+
+	var drop policy.DropPolicy
+	if r.Drop {
+		drop = policy.DropMust
+	}
+
+	return view.MappingRule{
+		ID:              id,
+		Name:            name,
+		Filter:          filter,
+		AggregationID:   aggID,
+		StoragePolicies: storagePolicies,
+		DropPolicy:      drop,
+	}, nil
+}
+
+// StoragePolicyConfiguration is the storage policy to apply to a set of metrics.
+type StoragePolicyConfiguration struct {
+	Retention  time.Duration `yaml:"retention"`
+	Resolution time.Duration `yaml:"resolution"`
+}
+
+// StoragePolicy returns the corresponding storage policy.
+func (p StoragePolicyConfiguration) StoragePolicy() (policy.StoragePolicy, error) {
+	return policy.ParseStoragePolicy(p.Resolution.String() + ":" + p.Retention.String())
+}
+
+// StoragePolicyConfigurations are a set of storage policy configurations.
+type StoragePolicyConfigurations []StoragePolicyConfiguration
+
+// StoragePolicies returns storage policies.
+func (p StoragePolicyConfigurations) StoragePolicies() (policy.StoragePolicies, error) {
+	var storagePolicies policy.StoragePolicies
+	for _, policy := range p {
+		value, err := policy.StoragePolicy()
+		if err != nil {
+			return nil, err
+		}
+		storagePolicies = append(storagePolicies, value)
+	}
+	return storagePolicies, nil
+}
+
+// RollupRuleConfiguration is a rollup rule configuration.
+type RollupRuleConfiguration struct {
+	// Filter is a string separated filter of labe name to label value
+	// glob patterns to filter the mapping rule to.
+	// e.g. "app:*nginx* foo:bar baz:qux*qaz*"
+	Filter string `yaml:"filter"`
+
+	// Transforms are a set of of rollup rule transforms.
+	Transforms []TransformConfiguration `yaml:"transforms"`
+
+	// StoragePolicies is the retention/resolution storage policies to keep the
```

Comment:
`StoragePolicies are the retention/resolution storage policies at which to keep the matched metrics`